### ISSUE TITLE
Higher level helper function to add background basemap to existing plot

### ISSUE THREAD

**Issue by jorisvandenbossche** [darribas/contextily#20]
From discussion on GeoPython, would there be interest in adding functionality along this lines?

```python
def add_basemap(ax, zoom, url='http://tile.stamen.com/terrain/tileZ/tileX/tileY.png'):
    xmin, xmax, ymin, ymax = ax.axis()
    basemap, extent = ctx.bounds2img(xmin, ymin, xmax, ymax, zoom=zoom, url=url)
    ax.imshow(basemap, extent=extent, interpolation='bilinear')
    # restore original x/y limits
    ax.axis((xmin, xmax, ymin, ymax))
```

I think something like this would make it easier to use `contextily` in practice to quickly add a map background to a plot.

The above now assumes you plotted on the ax in Web Mercator. In principle we could also accept a crs to also deal with other projections (but then you probably get even less quality basemap by transforming them).

cc @ljwolf

**Comment by jorisvandenbossche** [comment]
Example usage could be like:

```
neighbourhoods = geopandas.read_file('geo/geopython-pysal-tutorial/data/berlin-neighbourhoods.geojson').to_crs(epsg=3857)

ax = neighbourhoods.plot(figsize=(10, 10), alpha=0.5)
ctx.add_basemap(ax, zoom=10)
ax.set_axis_off()
```

giving

![image](https://user-images.githubusercontent.com/1020496/39785841-f37430bc-531d-11e8-9b5e-3310473bc0b3.png)

**Comment by jorisvandenbossche** [comment]
> contextily devs might want to raise/warn based on the howmany value, but this looks like a reasonable enhancement.

Yep, eg a maximum allowed value (that can be overriden with a keyword argument?)

**Comment by ljwolf** [comment]
This is what we're thinking for this function:

```python
def add_basemap(ax, zoom=’auto’, url = sources.ST_Terrain, 
			      interpolation=’sinc’,  attribution_text = ‘’, 
                             **extra_imshow_args)
    if zoom.lower() == 'auto':
        zoom = calculate_zoom()
    if url is filepath:
        import rasterio
        image = rasterio.read_and_cut_raster_to_axis_bbox(url)
    else:
        image = fetch_tile_from_provider(zoom,url,...)
    ax.imshow(image, interpolation=interpolation, **extra_imshow_args)
    return ax
```
where
- `ax`: matplotlib axes whose extent is set to be a compatible webmercator projection
- `zoom`: computed using `calculate_zoom`, but can be specified by the user if needed.
- `url`: a web address from `contextily.sources` or user-provided. Or, a filepath to a local image file. 
- `interpolation`: addressing #15 
- `attribution_text`: @darribas can elaborate
- `**extra_imshow_args`: for plotting/sending to imshow

**Comment by darribas** [comment]
This is now addressed in #32
